### ISSUE TITLE
make -j8 error while building from source

### ISSUE THREAD

**Issue by alwynmathew** [DLR-RM/granite#4]
Moved https://github.com/DLR-RM/granite/issues/3#issuecomment-1068253286 to new issues

OS: Ubuntu 18.04.6 LTS
Compiler: gcc (Ubuntu 7.5.0-3ubuntu1 18.04) 7.5.0

`make -j8` error while building from source

```
[100%] Built target test_nfr
/home/alwyn/granite/src/vio_sim.cpp: In function ‘void load_spline(std::__cxx11::string)’:
/home/alwyn/granite/src/vio_sim.cpp:775:8: error: ‘std::filesystem’ has not been declared
   std::filesystem::path path = spline_path;
        ^~~~~~~~~~
/home/alwyn/granite/src/vio_sim.cpp:777:13: error: ‘std::filesystem’ has not been declared
   if (!std::filesystem::exists(path)) {
             ^~~~~~~~~~
/home/alwyn/granite/src/vio_sim.cpp:777:32: error: ‘path’ was not declared in this scope
   if (!std::filesystem::exists(path)) {
                                ^~~~
/home/alwyn/granite/src/vio_sim.cpp:777:32: note: suggested alternative: ‘powh’
   if (!std::filesystem::exists(path)) {
                                ^~~~
                                powh
/home/alwyn/granite/src/vio_sim.cpp:782:42: error: ‘path’ was not declared in this scope
   std::cout << "Loading spline from " << path << std::endl;
                                          ^~~~
/home/alwyn/granite/src/vio_sim.cpp:782:42: note: suggested alternative: ‘powh’
   std::cout << "Loading spline from " << path << std::endl;
                                          ^~~~
                                          powh
/home/alwyn/granite/src/vio_sim.cpp:774:30: warning: unused parameter ‘spline_path’ [-Wunused-parameter]
 void load_spline(std::string spline_path) {
                              ^~~~~~~~~~~
In file included from /home/alwyn/granite/src/vio_sim.cpp:68:0:
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h: At global scope:
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:114:23: error: inline function ‘void granite::Se3SplineBase<Scalar>::knots_push_back(const SE3&) [with Scalar = double; granite::Se3SplineBase<Scalar>::SE3 = Sophus::SE3<double>]’ used but never defined [-Werror]
   virtual inline void knots_push_back(const SE3 &knot) = 0;
                       ^~~~~~~~~~~~~~~
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:90:36: error: inline function ‘int granite::Se3SplineBase<Scalar>::getDegree() [with Scalar = double]’ used but never defined [-Werror]
   [[nodiscard]] virtual inline int getDegree() = 0;
                                    ^~~~~~~~~
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:186:23: error: inline function ‘granite::Se3SplineBase<Scalar>::Vec3 granite::Se3SplineBase<Scalar>::transAccelWorld(int64_t) const [with Scalar = double; granite::Se3SplineBase<Scalar>::Vec3 = Eigen::Matrix<double, 3, 1>; int64_t = long int]’ used but never defined [-Werror]
   virtual inline Vec3 transAccelWorld(int64_t time_ns) const = 0;
                       ^~~~~~~~~~~~~~~
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:196:23: error: inline function ‘granite::Se3SplineBase<Scalar>::Vec3 granite::Se3SplineBase<Scalar>::rotVelBody(int64_t) const [with Scalar = double; granite::Se3SplineBase<Scalar>::Vec3 = Eigen::Matrix<double, 3, 1>; int64_t = long int]’ used but never defined [-Werror]
   virtual inline Vec3 rotVelBody(int64_t time_ns) const = 0;
                       ^~~~~~~~~~
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:191:23: error: inline function ‘granite::Se3SplineBase<Scalar>::Vec3 granite::Se3SplineBase<Scalar>::transVelWorld(int64_t) const [with Scalar = double; granite::Se3SplineBase<Scalar>::Vec3 = Eigen::Matrix<double, 3, 1>; int64_t = long int]’ used but never defined [-Werror]
   virtual inline Vec3 transVelWorld(int64_t time_ns) const = 0;
                       ^~~~~~~~~~~~~
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:151:29: error: inline function ‘const SO3& granite::Se3SplineBase<Scalar>::getKnotSO3(size_t) const [with Scalar = double; granite::Se3SplineBase<Scalar>::SO3 = Sophus::SO3<double>; size_t = long unsigned int]’ used but never defined [-Werror]
   virtual inline const SO3 &getKnotSO3(size_t i) const = 0;
                             ^~~~~~~~~~
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:163:30: error: inline function ‘const Vec3& granite::Se3SplineBase<Scalar>::getKnotPos(size_t) const [with Scalar = double; granite::Se3SplineBase<Scalar>::Vec3 = Eigen::Matrix<double, 3, 1>; size_t = long unsigned int]’ used but never defined [-Werror]
   virtual inline const Vec3 &getKnotPos(size_t i) const = 0;
                              ^~~~~~~~~~
cc1plus: all warnings being treated as errors
CMakeFiles/granite_vio_sim.dir/build.make:62: recipe for target 'CMakeFiles/granite_vio_sim.dir/src/vio_sim.cpp.o' failed
make[2]: *** [CMakeFiles/granite_vio_sim.dir/src/vio_sim.cpp.o] Error 1
CMakeFiles/Makefile2:107: recipe for target 'CMakeFiles/granite_vio_sim.dir/all' failed
make[1]: *** [CMakeFiles/granite_vio_sim.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by alwynmathew** [comment]
According to @hobbeshunter, I replaced all `std::filesystem` to `granite::fs` and added `#include <granite/utils/filesystem.h>` in `granite/src/vio_sim.cpp` but failed `make  -j8`:

```
[100%] Built target granite_calibrate
In file included from /home/alwyn/granite/src/vio_sim.cpp:68:0:
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:114:23: error: inline function ‘void granite::Se3SplineBase<Scalar>::knots_push_back(const SE3&) [with Scalar = double; granite::Se3SplineBase<Scalar>::SE3 = Sophus::SE3<double>]’ used but never defined [-Werror]
   virtual inline void knots_push_back(const SE3 &knot) = 0;
                       ^~~~~~~~~~~~~~~
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:90:36: error: inline function ‘int granite::Se3SplineBase<Scalar>::getDegree() [with Scalar = double]’ used but never defined [-Werror]
   [[nodiscard]] virtual inline int getDegree() = 0;
                                    ^~~~~~~~~
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:186:23: error: inline function ‘granite::Se3SplineBase<Scalar>::Vec3 granite::Se3SplineBase<Scalar>::transAccelWorld(int64_t) const [with Scalar = double; granite::Se3SplineBase<Scalar>::Vec3 = Eigen::Matrix<double, 3, 1>; int64_t = long int]’ used but never defined [-Werror]
   virtual inline Vec3 transAccelWorld(int64_t time_ns) const = 0;
                       ^~~~~~~~~~~~~~~
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:196:23: error: inline function ‘granite::Se3SplineBase<Scalar>::Vec3 granite::Se3SplineBase<Scalar>::rotVelBody(int64_t) const [with Scalar = double; granite::Se3SplineBase<Scalar>::Vec3 = Eigen::Matrix<double, 3, 1>; int64_t = long int]’ used but never defined [-Werror]
   virtual inline Vec3 rotVelBody(int64_t time_ns) const = 0;
                       ^~~~~~~~~~
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:191:23: error: inline function ‘granite::Se3SplineBase<Scalar>::Vec3 granite::Se3SplineBase<Scalar>::transVelWorld(int64_t) const [with Scalar = double; granite::Se3SplineBase<Scalar>::Vec3 = Eigen::Matrix<double, 3, 1>; int64_t = long int]’ used but never defined [-Werror]
   virtual inline Vec3 transVelWorld(int64_t time_ns) const = 0;
                       ^~~~~~~~~~~~~
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:151:29: error: inline function ‘const SO3& granite::Se3SplineBase<Scalar>::getKnotSO3(size_t) const [with Scalar = double; granite::Se3SplineBase<Scalar>::SO3 = Sophus::SO3<double>; size_t = long unsigned int]’ used but never defined [-Werror]
   virtual inline const SO3 &getKnotSO3(size_t i) const = 0;
                             ^~~~~~~~~~
/home/alwyn/granite/thirdparty/granite-headers/include/granite/spline/se3_spline.h:163:30: error: inline function ‘const Vec3& granite::Se3SplineBase<Scalar>::getKnotPos(size_t) const [with Scalar = double; granite::Se3SplineBase<Scalar>::Vec3 = Eigen::Matrix<double, 3, 1>; size_t = long unsigned int]’ used but never defined [-Werror]
   virtual inline const Vec3 &getKnotPos(size_t i) const = 0;
                              ^~~~~~~~~~
cc1plus: all warnings being treated as errors
CMakeFiles/granite_vio_sim.dir/build.make:62: recipe for target 'CMakeFiles/granite_vio_sim.dir/src/vio_sim.cpp.o' failed
make[2]: *** [CMakeFiles/granite_vio_sim.dir/src/vio_sim.cpp.o] Error 1
CMakeFiles/Makefile2:107: recipe for target 'CMakeFiles/granite_vio_sim.dir/all' failed
make[1]: *** [CMakeFiles/granite_vio_sim.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by hobbeshunter** [comment]
Thanks for reporting.

I could recreate it. Will try to look into it when I find time.

**Comment by hobbeshunter** [comment]
Should be fixed with [2380553](https://github.com/DLR-RM/granite/commit/23805532c52868d65607d10d33e14189d449e7fd)
